src/providers/trusted_service/context/asym_sign.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::ts_protobuf::{SignHashIn, SignHashOut, VerifyHashIn};
+use super::Context;
+use log::info;
+use parsec_interface::operations::psa_algorithm::AsymmetricSignature;
+use parsec_interface::requests::ResponseStatus;
+use std::convert::TryInto;
+
+impl Context {
+    pub fn asym_sign(
+        &self,
+        key_id: u32,
+        hash: Vec<u8>,
+        algorithm: AsymmetricSignature,
+    ) -> Result<Vec<u8>, ResponseStatus> {
+        info!("Handling GenerateKey request");
+        let mut proto_req = SignHashIn {
+            handle: 0,
+            hash,
+            alg: algorithm.try_into()?,
+        };
+        let SignHashOut { signature } = self.send_request_with_key(&mut proto_req, key_id)?;
```

Comment:
Maybe the message can be given by ownership to `send_request_with_key` instead of reference? I guess you would not need to use the request structure afterwards.

Reply:
True, will change